plugins/components/collider/Collider.hpp

```diff
@@ -0,0 +1,143 @@
+/*
+** EPITECH PROJECT, 2024
+** RType
+** File description:
+** Collider.hpp file
+*/
+
+#ifndef COLLIDER_HPP
+#define COLLIDER_HPP
+
+#include "plugins/components/AComponent.hpp"
+#include "GameEngine/GameEngine.hpp"
+#include <vector>
+#include <stdexcept>
+#include <cstdint>
+#include <libconfig.h++>
+
+namespace Components {
+
+    class Collider : public AComponent {
+    public:
+        /**
+         * @brief Default constructor
+         * 
+         * Initializes the rectangle collider with default width and height of 0.
+         */
+        Collider(float width = 0.0f, float height = 0.0f) : AComponent(std::string("Collider")), width(width), height(height) {};
+
+        /**
+         * @brief Constructor using configuration.
+         * 
+         * Initializes the rectangle collider based on config settings.
+         */
+        Collider(libconfig::Setting &config) : AComponent("Collider") {
+            config.lookupValue("width", width);
```

Comment:
check if the lookupvalue fails and if it does, give default values to width and height